File: src/modules/Pim.Module.Stats/Entities/AppUsageEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Pim.Module.Stats.Entities;

[Table("app_usage")]
public class AppUsageEntity
{
    [Key][Column("id")] public long Id { get; set; }
    [Column("device_id")][MaxLength(64)] public string DeviceId { get; set; } = string.Empty;
    [Column("package_name")][MaxLength(256)] public string PackageName { get; set; } = string.Empty;
    [Column("start_time")] public DateTimeOffset StartTime { get; set; }
    [Column("end_time")] public DateTimeOffset EndTime { get; set; }
    [Column("duration_ms")] public long DurationMs { get; set; }
    [Column("last_time_used")] public DateTimeOffset LastTimeUsed { get; set; }
    [Column("created_at")] public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}

File: src/client-android/app/src/main/java/com/pim/app/data/AppUsageEntity.kt
package com.pim.app.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "app_usage")
data class AppUsageEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    @ColumnInfo(name = "package_name") val packageName: String,
    @ColumnInfo(name = "start_time") val startTime: Long,
    @ColumnInfo(name = "end_time") val endTime: Long,
    @ColumnInfo(name = "duration_ms") val durationMs: Long,
    @ColumnInfo(name = "last_time_used") val lastTimeUsed: Long,
    @ColumnInfo(name = "synced") val synced: Boolean = false
)
